### DISCUSSION TITLE
Geographics, Travel, Maps, Geocaching

### DISCUSSION THREAD

**Discussion by gsantner** [gsantner/markor#1739] [Ideas]
## General information

* **App version:** v2.1
* **System:** v9

### Read this first
* Please share first some ideas, what you want to do and what do you think would be nice. **This is not a topic that can be "fixed" or plain implemented down.**
* For ideas - it's really ideas, it may or may not be possible that libraries or other apps return said data or have said functionalities. Do research if it's possible.
* Markor + Privacy
  * Don't add LOCATION / GPS permission to Markor
  * No Internet, complete offline usage. Interaction with webservices is not allowed (e.g. geocorders).

---------------

---------------

### What is this about?
* To add features to make Markor attractive for writing documents that are about places in the world
* This includes but is not limited to
  * Travel planning and taking travel notes
  * Use in school/education for geographics classes
  * Cycle and walking trips with friends
  * Noting favourite places, bars, restaurants, swim places etc
  * Geocaching

---------------

### Ideas
* Markdown support for `geo:lat,lng`
  * Note: Available. See flexmark-java wiki
* Pick location with other app
  * Start maps app with Intent to pick a location (OsmAND, Google Maps)
  * Add coordinates to document
  * Add place details (like address or POI name) to document
* TextAction to open Maps app
   * To show a (1) place in maps app
   * To start navigation to a (1) place
   * To open a route (multiple locations)
* Draw points on a static image of the world map
  * Add a image file (jpg/png) to Markor
  * Find coordinates in current document
  * Draw a dot on each coordinate at the world map

---------------

**Comment by sudoalgorithm** [comment]
I would love to implement the second idea i.e Pick a location with other apps If no one is doing that right now.

**Comment by gsantner** [comment]
@sudoalgorithm 
Thanks for interest, and feel free to go. Please however note that theres more than just that point :smile: . Also take other stuff with if you want to. You can also bring up own ideas if you have. 

As far possible, please try to write code in the `CoolExperiments.java` file

**Comment by gsantner** [comment]
@sudoalgorithm 
have you started / stopped?

**Comment by sudoalgorithm** [comment]
@gsantner working on it right now 😅

**Comment by gsantner** [comment]
there was no outcome, right?

**Comment by gsantner** [comment]
Just hidden comments as there was never something coming in :/

**Comment by k3b** [comment]
if somebody is interested: here is a description how interaction with geo is implemented in android: https://github.com/k3b/k3b-geoHelper/wiki/Android-Geo-howto

Here are some geo-aware open source android apps: https://github.com/k3b/k3b-geoHelper/wiki/Android-Geo-aware-apps
